File: backend-bot/src/main/java/cc/jktu/apps/bot/service/BotService.java
package cc.jktu.apps.bot.service;

import lombok.Getter;
import net.mamoe.mirai.Bot;
import net.mamoe.mirai.event.ListenerHost;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class BotService {

    private final List<ListenerHost> eventHandlers;

    @Getter
    private final Bot bot;

    public BotService(List<ListenerHost> eventHandlers, Bot bot) {
        this.eventHandlers = eventHandlers;
        this.bot = bot;
    }

    public void registerHandlers() {
        eventHandlers.forEach(bot.getEventChannel()::registerListenerHost);
    }

}

File: backend-bot/src/main/kotlin/cc/jktu/apps/bot/service/BotService.kt
package cc.jktu.apps.bot.service

import net.mamoe.mirai.Bot
import net.mamoe.mirai.event.ListenerHost
import org.springframework.stereotype.Service

@Service
class BotService(private val eventHandlers: List<ListenerHost>, val bot: Bot) {

    fun registerHandlers() {
        eventHandlers.forEach {
            bot.eventChannel.registerListenerHost(it)
        }
    }
}
